pkg/terminal/command.go

```diff
@@ -1341,6 +1351,175 @@ func edit(t *Term, ctx callContext, args string) error {
+		for i, v := range arr {
+			str += "\"" + string(v) + "\""
+			if i != len(arr)-1 {
+				str += " "
+			}
+		}
+		str += "]"
+		return str
+	}
+
+	// if no format and size, use default argument 'xb'
+	if i == len(fmtStr) {
+		return count, 'x', 1, nil
+	}
+
+	supportedFormat := []byte{'o', 'x', 'd'}
+	format = fmtStr[i]
+	hitSupportedFlag := false
+	for _, v := range supportedFormat {
+		if v == format {
+			hitSupportedFlag = true
+			break
+		}
+	}
+	if !hitSupportedFlag {
+		return 0, ' ', 0, fmt.Errorf("the \"format\" letter should be one of %s", formatSupportedArrForErr(supportedFormat))
+	}
+
+	i++
+	// if no size, use default argument 'b'
+	if i == len(fmtStr) {
+		return count, format, 1, nil
+	}
+	supportedSize := []byte{'b', 'h', 'w', 'g'}
+	hitSupportedFlag = false
+	var index int
+	for index = 0; index < len(supportedSize); index++ {
+		if supportedSize[index] == fmtStr[i] {
+			hitSupportedFlag = true
+			break
+		}
+	}
+	if !hitSupportedFlag {
+		return 0, ' ', 0, fmt.Errorf("the \"size\" letter should be one of %s", formatSupportedArrForErr(supportedSize))
+	}
+	size = 1 << uint(index)
+
+	i++
+	// if redundant parameters, should report error
+	if i != len(fmtStr) {
+		return 0, ' ', 0, fmt.Errorf("some redundant parameters for fmt")
+	}
+	return count, format, size, nil
+}
+
+func examineMemoryCmd(t *Term, ctx callContext, args string) error {
+	v := strings.FieldsFunc(args, func(c rune) bool {
+		return c == ' '
+	})
+	if len(v) != 1 && len(v) != 3 && len(v) != 5 {
+		return fmt.Errorf("wrong arguments for input")
+	}
+	// Default value
+	format := "hex"
+	length := 1
+	addressStr := v[len(v)-1]
+
+	StringArrIndex := func(strArr []string, str string) int {
+		for i, v := range strArr {
+			if v == str {
+				return i
+			}
+		}
+		return -1
+	}
+
+	fmtIdx := StringArrIndex(v, "-fmt")
+	if fmtIdx != -1 {
+		if fmtIdx == len(v)-1 {
+			return fmt.Errorf("a format must be followed with -fmt")
+		}
+		if fmtIdx != 0 && fmtIdx != 2 {
+			return fmt.Errorf("wrong arugments for input")
+		}
+		format = v[fmtIdx+1]
+	}
+
+	fmtMapToPriFmt := map[string]byte{
+		"oct":         'o',
+		"octal":       'o',
+		"hex":         'x',
+		"hexadecimal": 'x',
+		"dec":         'd',
+		"decimal":     'd',
+		"bin":         'b',
+		"binary":      'b',
+	}
+	priFmt, ok := fmtMapToPriFmt[format]
+	if !ok {
+		return fmt.Errorf("a wrong format argument followed with -fmt")
+	}
+
+	lenIdx := StringArrIndex(v, "-len")
+	if lenIdx != -1 {
+		if lenIdx == len(v)-1 {
+			return fmt.Errorf("a length must be followed with -len")
+		}
+		if lenIdx != 0 && lenIdx != 2 {
+			return fmt.Errorf("wrong arguments for input")
+		}
+		var err error
+		length, err = strconv.Atoi(v[lenIdx+1])
+		if err != nil {
+			return fmt.Errorf("len must be an positive integer")
+		}
+		if length <= 0 {
+			return fmt.Errorf("len must be an positive integer")
+		}
+	}
+
+	// TODO, maybe configured by user.
+	if length > 1000 {
+		return fmt.Errorf("len must be not more than 1k")
```

Comment:
"len must be less than 1000"

Reply:
I use `not more than` because `length` belong to [1, 1000] instead of [1, 1000). 
So maybe "len must be less than or equal to 1000“ better, what's your options?